core/configuration/dt_config_provider.go

```diff
@@ -68,6 +68,13 @@ func (config *DtConfiguration) TenantId() int32 {
 	return config.tenantId
 }
 
+func (config *DtConfiguration) QualifiedTenantId() QualifiedTenantId {
```

Comment:
Instead of providing a getter, this could be added directly to the configuration I think (I think there was some rather commonly used annotation like `mapstructure squash` to avoid the additional nesting level in the YAML/JSON)

Reply:
I found that adding a getter here caused the least friction, especially wrt. to additional requirements the other approach would add for serialization as you mentioned. If you don't mind, I'd leave it as-is for now.